exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +74,2825 @@ func TestStart(t *testing.T) {
+		t.Errorf("%s GetSubAccounts() error %v", g.Name, err)
+	}
+}
+
+func TestGetSingleSubAccount(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetSingleSubAccount(context.Background(), "Sub_Account_for_testing"); err != nil && !strings.Contains(err.Error(), "FORBIDDEN") {
+		t.Errorf("%s GetSingleSubAccount() error %v", g.Name, err)
+	}
+}
+
+// Wrapper test functions
+
+func TestFetchTradablePairs(t *testing.T) {
+	t.Parallel()
+	_, err := g.FetchTradablePairs(context.Background(), asset.DeliveryFutures)
+	if err != nil {
+		t.Errorf("%s FetchTradablePairs() error %v", g.Name, err)
+	}
+	if _, err := g.FetchTradablePairs(context.Background(), asset.Options); err != nil {
+		t.Errorf("%s FetchTradablePairs() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateTradablePairs(t *testing.T) {
+	t.Parallel()
+	if err := g.UpdateTradablePairs(context.Background(), true); err != nil {
+		t.Errorf("%s UpdateTradablePairs() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateTickers(t *testing.T) {
+	t.Parallel()
+	if err := g.UpdateTickers(context.Background(), asset.DeliveryFutures); err != nil {
+		t.Errorf("%s UpdateTickers() error %v", g.Name, err)
+	}
+}
+
+func TestUpdateOrderbook(t *testing.T) {
+	t.Parallel()
+	cp, err := getFirstTradablePair(t, asset.Futures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := g.UpdateOrderbook(context.Background(), cp, asset.Futures); err != nil {
+		t.Errorf("%s UpdateOrderbook() error %v", g.Name, err)
+	}
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.Skip(credInformationNotProvided)
+	}
+	if _, err := g.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Empty); err != nil {
+		t.Errorf("%s GetWithdrawalsHistory() error %v", g.Name, err)
+	}
+}
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	currencyPair := currency.NewPair(currency.BTC, currency.USDT)
+	_, err := g.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	cp, err := getFirstTradablePair(t, asset.DeliveryFutures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), cp, asset.DeliveryFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), currencyPair, asset.Options)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip(credInformationNotProvidedOrManipulatingRealOrdersNotAllowed)
+	}
+	var orderSubmission = &order.Submit{
+		Exchange: g.Name,
+		Pair: currency.Pair{
+			Delimiter: currency.UnderscoreDelimiter,
+			Base:      currency.BTC,
+			Quote:     currency.USDT,
+		},
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		AssetType: asset.CrossMargin,
+	}
+	_, err := g.SubmitOrder(context.Background(), orderSubmission)
```

Comment:
This prior [nit](https://github.com/thrasher-corp/gocryptotrader/pull/1058/files#r1051709868) Is still not complete, yet has a thumbs up.